### PR TITLE
Better Python3 compatibility

### PR DESCRIPTION
**Pull request by brot**
Based on the discussion in https://github.com/arskom/spyne/issues/632 I fixed a few test. Looked into a few more, but not sure how to fix them.

Additional question:
Are there any considerations dropping Python2.x support, because Python2 is nearly end of life now. https://www.python.org/dev/peps/pep-0373/

### CODE REVIEW

**Comment by bgermann** [comment]
See also #628.

**Comment by plq** [comment]
Jenkins, run all tests.

File: spyne/protocol/dictdoc/hier.py [hunk 1]
```diff
@@ -318,6 +318,10 @@ def _doc_to_object(self, ctx, cls, doc, validator=None):
 
         # parse input to set incoming data to related attributes.
         for k, v in items:
+            if not six.PY2:
+                if isinstance(k, bytes):
+                    k = k.decode('utf8')
+
```
Comment (plq) [inline spyne/protocol/dictdoc/hier.py]: we are not supposed to hardcode encodings like this

File: spyne/test/protocol/test_msgpack.py [hunk 2]
```diff
@@ -111,12 +111,19 @@ def get_values(ctx, keys):
...
+                {"KeyValuePair": {'value': 'd', 'key': 'c'}},
+            ]
+        else:
+            values_result = [
+                {"KeyValuePair": {'key': 'a', 'value': 'b'}},
+                {"KeyValuePair": {'key': 'c', 'value': 'd'}},
+            ]
+
+        s = msgpack.packb([1, 0, None, {'get_valuesResponse':
+            {'get_valuesResult': values_result}}
```
Comment (plq) [inline spyne/test/protocol/test_msgpack.py]: if this works at all, it's pure luck.

**Pull request closed without merging**